src/Fable.Transforms/FableTransforms.fs

```diff
@@ -275,7 +275,8 @@ module private Transforms =
             | value when ident.IsCompilerGenerated
                     // Don't erase the binding if the compiler-generated ident is a tuple, because the getters
                     // will be erased later (see above) and there's a risk the expression gets totally removed
-                    && not(isTuple ident.Type)
+                    && not (isTuple ident.Type)
+                    && not (ident.Name.StartsWith("patternInput")) // tuple may be split into multiple getters
```

Comment:
The line before is already checking whether the ident is a tuple. Isn't this redundant? I'd prefer not to have too many conditions for the optimizations as this complicates maintenance. Do you have an example of when this condition is necessary (and the one above is not enough)?

Reply:
@alfonsogarciacaro No, it's not redundant. This catches the case when a tuple got split into multiple separate getters (so isTuple is false, but we still want to recognize it).

Reply:
Ah, ok. Thanks for confirming. Does it only happen in the optimized AST? Would it be possible to add a test case to avoid regressions? I'm concerned that the "patternInput" literal is an implementation detail from FCS that could change at any time.

Reply:
@alfonsogarciacaro Yes, it is an implementation detail, but so is everything else in FCS when code is generated or transformed before building the AST. I'll try to add a test when I have some free time, but in the mean time can we leave it there, if it doesn't break anything for you?

Reply:
Sure :)